src/MongoDB.Driver.Core/Core/Authentication/SaslPrepHelper.cs

```diff
@@ -100,7 +100,7 @@ internal static string SaslPrepStored(string str)
 #if NET45
             var normalized = new string(chars.Take(length).ToArray()).Normalize(NormalizationForm.FormKC);
 #else
-            var normalized = new string(chars);
+            var normalized = new string(chars.Take(length).ToArray());
```

Comment:
We can't normalize on .NET Standard 1.6 without taking a dependency. As it stands, there's a chance that simple passwords will work on .NET Standard, especially if the password is normalized ahead of time.